test/kwtool_web/controllers/keyword_controller_test.exs

```diff
@@ -24,7 +27,22 @@ defmodule KwtoolWeb.KeywordControllerTest do
         |> with_signed_in_user(created_user)
         |> get(Routes.keyword_path(conn, :show, keyword.id))
 
-      assert html_response(conn, 200) =~ "Result for"
+      assert html_response(conn, 200) =~ "Result for \"#{keyword.phrase}\""
+    end
+
+    test "redirects to the keywords listing page when given a keyword does NOT belong to the logged-in user",
+         %{conn: conn} do
+      created_user_1 = insert(:user)
+
+      created_user_2 = insert(:user)
+      keyword_of_user_2 = insert(:keyword, user: created_user_2)
+
+      conn =
+        conn
+        |> with_signed_in_user(created_user_1)
+        |> get(Routes.keyword_path(conn, :show, keyword_of_user_2.id))
+
+      assert redirected_to(conn) == Routes.keyword_path(conn, :index)
```

Comment:
Should we assert that there should be a flash `Keyword not found.` 💡

Reply:
Updated! Thanks 👍